### PR TITLE
Simplify error macros

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Error macros are too complicated, passing `context: ` is not required. This PR adjusts this

## 🚫 Blocked by

- ~#590~

## 🎥  Demo

**Before**:
```rust
report!(context: std::io::Error { .. }, "Some error message");
```

**After** (Will be forbidden in a follow up PR as string literals will be removed from macros):
```rust
report!("Some error message").provide_context(std::io::Error { .. });
```
---

**Before**:
```rust
let report = report!(context: std::io::Error { .. });
assert!(report.contains::<std::io::Error>);

let report = report!(std::io::Error { .. });
assert!(report.contains::<std::io::Error>); // Fails
```

**After**:
```rust
let report = report!(std::io::Error { .. });
assert!(report.contains::<std::io::Error>); // Succeeds
```

### CODE REVIEW

File: packages/engine/lib/error/src/frame.rs [hunk 1]
```diff
@@ -169,10 +177,7 @@ impl FrameRepr {
         unsafe { (self.vtable.object_ref)(self) }
     }
 
-    fn downcast<C>(&self) -> Option<NonNull<C>>
-    where
-        C: Context,
-    {
+    fn downcast<C: Any>(&self) -> Option<NonNull<C>> {
         let target = TypeId::of::<C>();
         // Use vtable to attach C's native vtable for the right original type C.
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/error/src/frame.rs:182]: Can you add a safety by the way?

Comment (TimDiekmann) [reply packages/engine/lib/error/src/frame.rs]: I enforced this for this crate: 5df6e3a3a2029dd2e375a5fb5e7dec6b0ec49adf

File: packages/engine/lib/error/src/macros.rs [hunk 2]
```diff
@@ -5,20 +5,18 @@ pub mod __private {
...
-        use core::marker::PhantomData;
+        #![allow(clippy::unused_self)]
 
-        use crate::{Message, Report};
+        use crate::{Context, Message, Report};
 
-        pub trait AdhocKind: Sized {
-            fn __kind(&self) -> Adhoc {
-                Adhoc
+        pub trait MessageTag {
```
Comment (TimDiekmann) [inline packages/engine/lib/error/src/macros.rs:12]: I will add documentation on those "black magic" when this got cleaned up. I expect changes here in a follow-up PR.

**Comment by TimDiekmann** [comment]
Rebased onto `main` and added a Demo

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**